client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -304,31 +295,17 @@ const UnifiedDesignPickerStep: StepType< {
 					},
 					optionalProps
 				);
-			} else if ( ! isSiteRequired && ! siteSlugOrId && _selectedDesign ) {
-				handleSubmit(
-					{
-						selectedDesign: _selectedDesign,
-						selectedSiteCategory: categorization.selections?.join( ',' ),
-					},
-					optionalProps
-				);
 			}
 		},
 		[
 			activateDesign,
-			assembleSite,
-			categorization.selections,
 			designs,
 			globalStyles,
 			handleSubmit,
-			isSiteRequired,
-			reduxDispatch,
 			selectedDesign,
 			selectedStyleVariation,
-			setDesignOnSite,
 			setPendingAction,
 			setSelectedDesign,
-			site?.ID,
```

Comment:
These dependencies were not used (probably leftover from a previous cleanup), and therefore we should be able to safely remove them